blamo/lib/Test/test.dart

```diff
@@ -150,21 +150,27 @@ class _TestPageState extends State<TestPage> {
                               ),
                               FormBuilderTextField(
                                 textInputAction: TextInputAction.next,
+                                keyboardType: TextInputType.number,
                                 focusNode: formNodes[1],
                                 attribute: 'beginTestDepth',
-                                validators: [FormBuilderValidators.numeric()],
+                                validators: [FormBuilderValidators.numeric(), FormBuilderValidators.max(0)],
                                 decoration: InputDecoration(labelText: "Begin Test Depth (-m)"),
                                 initialValue: formatValue(testObjectToBuildFrom.beginTest.toString()),
                                 onChanged: (void nbd){updateTestObject();},
                                 onFieldSubmitted: (v){
                                   FocusScope.of(context).requestFocus(formNodes[2]);
                                 },
-                                ),
+                              ),
                               FormBuilderTextField(
                                 textInputAction: TextInputAction.next,
+                                keyboardType: TextInputType.number,
                                 focusNode: formNodes[2],
                                 attribute: 'endTestDepth',
-                                validators: [FormBuilderValidators.numeric()],
+                                validators: [FormBuilderValidators.numeric(), FormBuilderValidators.max(0), (endDepth){
+                                  if(_fbKey.currentState != null && endDepth != null && _fbKey.currentState.fields["beginTestDepth"].currentState.value != null && double.tryParse(_fbKey.currentState.fields["beginTestDepth"].currentState.value) != null && double.tryParse(endDepth) != null && double.tryParse(endDepth) > double.tryParse(_fbKey.currentState.fields["beginTestDepth"].currentState.value))
```

Comment:
add >= to make the evaluation with the case of the values being equal when they should not be allowed to.